### PR TITLE
Alternative of General support for disguised types

### PR DESCRIPTION
**Pull request by EPashkin**
Part of #529
Followup of https://github.com/gtk-rs/gir/pull/544

cc @GuillaumeGomez,  @sdroege, @tmiasko

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -295,39 +295,17 @@ impl Library {
...
-                    target_c_type,
-                    doc: None, //TODO: temporary
-                })
-            );
-        }
+        // A 'disguised' structure is one where the c:type is a typedef that
+        // doesn't look like a pointer, but is internally: typedef struct _X *X;
+        //
+        // Currently g-ir-sanner is too eager to mark all typedef to non-complete
+        // types as disguised, so we limit this behaviour two most common cases.
```
Comment (GuillaumeGomez) [inline src/parser.rs]: "two" -> "to"?

File: src/parser.rs [hunk 2]
```diff
@@ -295,39 +295,17 @@ impl Library {
...
-                    typ: tid,
-                    target_c_type,
-                    doc: None, //TODO: temporary
-                })
-            );
-        }
+        // A 'disguised' structure is one where the c:type is a typedef that
+        // doesn't look like a pointer, but is internally: typedef struct _X *X;
+        //
+        // Currently g-ir-sanner is too eager to mark all typedef to non-complete
```
Comment (GuillaumeGomez) [inline src/parser.rs]: "sanner"?

File: src/parser.rs (hunk 1 above)
Comment (EPashkin) [reply src/parser.rs]: 2

**Comment by GuillaumeGomez** [comment]
👍 (and thanks for removing all extra whitespaces!)

File: src/parser.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/parser.rs]: Then it's "to the two". 😃

File: src/parser.rs (hunk 2 above)
Comment (EPashkin) [reply src/parser.rs]: Hm, Don't know.
@tmiasko It must be "g-ir-scanner" ?

**Comment by EPashkin** [comment]
Not all whitespaces, only in files that I touch.

**Comment by GuillaumeGomez** [comment]
Let me reword my sentence then: "and thanks for removing all extra whitespaces in those files!". Better? 😆

**Comment by EPashkin** [comment]
Do not thank me, the editor (emacs) does it himself while saving ;)

**Comment by GuillaumeGomez** [comment]
Just take the compliment and stop debating! Can't even thank people anymore now...

File: src/parser.rs (hunk 1 above)
Comment (EPashkin) [reply src/parser.rs]: Thanks, I will fix

File: src/parser.rs (hunk 2 above)
Comment (tmiasko) [reply src/parser.rs]: Yes, it should be g-ir-scanner. I made so many typos there, and now you have to fix them ;-).

**Comment by EPashkin** [comment]
Yes sir!

File: src/parser.rs [hunk 3]
```diff
@@ -295,39 +295,17 @@ impl Library {
...
-                    name: disguised_record_name.clone(),
-                    c_identifier: disguised_c_type,
-                    typ: tid,
-                    target_c_type,
-                    doc: None, //TODO: temporary
-                })
-            );
-        }
+        // A 'disguised' structure is one where the c:type is a typedef that
+        // doesn't look like a pointer, but is internally: typedef struct _X *X;
```
Comment (tmiasko) [inline src/parser.rs]: What about using this as a comment for Record::disguised field?
Seems like a more appropriate place now.

Comment (EPashkin) [reply src/parser.rs]: Good idea, it will be first comment in `library::Record` :wink:

**Pull request merged**